sixtyfps_runtime/rendering_backends/gl/event_loop.rs

```diff
@@ -268,6 +265,11 @@ pub(crate) static GLOBAL_PROXY: once_cell::sync::OnceCell<
     std::sync::Mutex<GlobalEventLoopProxyOrEventQueue>,
 > = once_cell::sync::OnceCell::new();
 
+#[cfg(target_arch = "wasm32")]
+thread_local! {
+    pub(crate) static GLOBAL_PROXY: RefCell<Option<GlobalEventLoopProxyOrEventQueue>> = RefCell::new(None)
```

Comment:
why does it have to be a thread_local and can't be a mutex like in other platforms?

Reply:
Because `EventLoopProxy` in winit is not `Send` in the wasm build:

```
error[E0277]: `Rc<winit::platform_impl::platform::event_loop::runner::Execution<CustomEvent>>` cannot be sent between threads safely
   --> sixtyfps_runtime/rendering_backends/gl/event_loop.rs:263:1
    |
263 | / pub(crate) static GLOBAL_PROXY: once_cell::sync::OnceCell<
264 | |     std::sync::Mutex<GlobalEventLoopProxyOrEventQueue>,
265 | | > = once_cell::sync::OnceCell::new();
    | |_____________________________________^ `Rc<winit::platform_impl::platform::event_loop::runner::Execution<CustomEvent>>` cannot be sent between threads safely
    |
    = help: within `GlobalEventLoopProxyOrEventQueue`, the trait `Send` is not implemented for `Rc<winit::platform_impl::platform::event_loop::runner::Execution<CustomEvent>>`
    = note: required because it appears within the type `winit::platform_impl::platform::event_loop::runner::Shared<CustomEvent>`
    = note: required because it appears within the type `winit::platform_impl::platform::event_loop::proxy::Proxy<CustomEvent>`
    = note: required because it appears within the type `EventLoopProxy<CustomEvent>`
note: required because it appears within the type `GlobalEventLoopProxyOrEventQueue`
   --> sixtyfps_runtime/rendering_backends/gl/event_loop.rs:229:17
    |
229 | pub(crate) enum GlobalEventLoopProxyOrEventQueue {
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: required because of the requirements on the impl of `Sync` for `Mutex<GlobalEventLoopProxyOrEventQueue>`
    = note: 1 redundant requirement hidden
    = note: required because of the requirements on the impl of `Sync` for `once_cell::imp::OnceCell<Mutex<GlobalEventLoopProxyOrEventQueue>>`
    = note: required because it appears within the type `once_cell::sync::OnceCell<Mutex<GlobalEventLoopProxyOrEventQueue>>`
    = note: shared static variables must have a type that implements `Sync`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `sixtyfps-rendering-backend-gl` due to previous error
```

This is a bit in contrary do the docs, but OTOH I recall that it's not just about changing one `Rc` to `Arc` inside winit - the wasm backend uses Rc throughout. The last time I tried I got several pages full of missing `Send` for various types. I can try again, but I'm not sure it's worth it given the actual single-threaded nature.